stixpy/frames.py

```diff
@@ -0,0 +1,174 @@
+
+    Parameters
+    ----------
+    obstime : `astropy.time.Time`
+        Time
+    Returns
+    -------
+    """
+    roll, pitch, yaw, sas_x, sas_y, solo_position_heeq = _get_aux_data(obstime)
+    if np.isnan(sas_x) or np.isnan(sas_y):
+        warnings.warn('SAS solution not available using spacecraft pointing and average SAS offset')
+        sas_x = yaw
+        sas_y = pitch
+
+    # Generate the rotation matrix using the x-convention (see Goldstein)
+    axes_matrix = np.array([[1, 0, 0],
+                            [0, 0, 1],
+                            [0, 1, 0]])
+    C = rotation_matrix(sas_x + STIX_X_OFFSET, "z")
+    B = rotation_matrix(-1*sas_y + STIX_Y_OFFSET, "y")
+    A = rotation_matrix(-1*roll, 'x')
+    rmatrix = matrix_product(A, B, C, axes_matrix)
+
+    return rmatrix, solo_position_heeq
+
+
+def _get_aux_data(obstime):
+    # Find, download, read aux file with pointing, sas and position information
+    logger.debug('Searching for AUX data')
+    query = Fido.search(a.Time(obstime, obstime), a.Instrument.stix, a.Level.l2,
+                        a.stix.DataType.aux, a.stix.DataProduct.aux_auxiliary)
+    if len(query['stix']) != 1:
+        raise ValueError('Exactly one AUX file should be found but %d were found.',
+                         len(query['stix']))
+    logger.debug('Downloading AUX data')
+    files = Fido.fetch(query)
+    if len(files.errors) > 0:
+        raise ValueError('There were errors downloading the data.')
+    # Read and extract data
+    logger.debug('Loading and extracting AUX data')
+    hdu = fits.getheader(files[0], ext=0)
+    aux = QTable.read(files[0], hdu=2)
+    start_time = Time(hdu.get('DATE-BEG'))
+    rel_date = (obstime - start_time).to('s')
+    idx = np.argmin(np.abs(rel_date - aux['time']))
+    sas_x, sas_y = aux[idx][['y_srf', 'z_srf']]
+    roll, yaw, pitch = aux[idx]['roll_angle_rpy']
+    solo_position_heeq = aux[idx]['solo_loc_heeq_zxy']
+    logger.debug('SAS: %s, %s, Orientation: %s, %s, %s, Position: %s', sas_x, sas_y, roll, yaw,
+                 pitch, solo_position_heeq)
+    return roll, pitch, yaw, sas_x, sas_y, solo_position_heeq
+
+
+@frame_transform_graph.transform(coord.FunctionTransform, STIXImagingFrame, Helioprojective)
+def stixim_to_hpc(stxcoord, hpcframe):
+    r"""
+    Transform STIX Imaging coordinates to given HPC frame
+    """
+    logger.debug('STIX: %s', stxcoord)
+
+    rot_matrix, solo_pos_heeq = _get_rotation_matrix_and_position(stxcoord.obstime)
+    solo_heeq = HeliographicStonyhurst(solo_pos_heeq, representation='cartesian',
+                                       obstime=stxcoord.obstime)
+
+    # Transform from STIX imaging to SOLO HPC
+    newrepr = stxcoord.cartesian.transform(rot_matrix)
+
+    # Create SOLO HPC
+    solo_hpc = Helioprojective(newrepr, obstime=stxcoord.obstime,
+                               observer=solo_heeq.transform_to(HeliographicStonyhurst))
```

Comment:
Hum it seem's I still need it running in python3.7 with astropy 4.2.1, sunpy 3.1.7? Otherwise I get an attribute error
`E       AttributeError: 'HeliographicStonyhurst' object has no attribute 'lat'`

Reply:
Actually same on python 3.8 with astropy 5.1 and sunpy 4.0.2